src/Admin/AbstractAdmin.php

```diff
@@ -1768,79 +1768,7 @@ final public function hasAccess(string $action, ?object $object = null): bool
      */
     final public function getActionButtons(string $action, ?object $object = null): array
     {
-        // nothing to do for non-internal actions
-        if (!isset(self::INTERNAL_ACTIONS[$action])) {
-            return [];
-        }
-
-        $actionBit = self::INTERNAL_ACTIONS[$action];
-
-        $buttonList = [];
-
-        if (0 !== (self::MASK_OF_ACTION_CREATE & $actionBit)
-            && $this->hasRoute('create')
-            && $this->hasAccess('create')
-        ) {
-            $buttonList['create'] = [
-                'template' => $this->getTemplateRegistry()->getTemplate('button_create'),
-            ];
-        }
-
-        $canAccessObject = 0 !== (self::MASK_OF_ACTIONS_USING_OBJECT & $actionBit)
-            && null !== $object
-            && null !== $this->id($object);
-
-        if ($canAccessObject
-            && 0 !== (self::MASK_OF_ACTION_EDIT & $actionBit)
-            && $this->hasRoute('edit')
-            && $this->hasAccess('edit', $object)
-        ) {
-            $buttonList['edit'] = [
-                'template' => $this->getTemplateRegistry()->getTemplate('button_edit'),
-            ];
-        }
-
-        if ($canAccessObject
-            && 0 !== (self::MASK_OF_ACTION_HISTORY & $actionBit)
-            && $this->hasRoute('history')
-            && $this->hasAccess('history', $object)
-        ) {
-            $buttonList['history'] = [
-                'template' => $this->getTemplateRegistry()->getTemplate('button_history'),
-            ];
-        }
-
-        if ($canAccessObject
-            && 0 !== (self::MASK_OF_ACTION_ACL & $actionBit)
-            && $this->isAclEnabled()
-            && $this->hasRoute('acl')
-            && $this->hasAccess('acl', $object)
-        ) {
-            $buttonList['acl'] = [
-                'template' => $this->getTemplateRegistry()->getTemplate('button_acl'),
-            ];
-        }
-
-        if ($canAccessObject
-            && 0 !== (self::MASK_OF_ACTION_SHOW & $actionBit)
-            && $this->hasRoute('show')
-            && $this->hasAccess('show', $object)
-            && \count($this->getShow()) > 0
-        ) {
-            $buttonList['show'] = [
-                'template' => $this->getTemplateRegistry()->getTemplate('button_show'),
-            ];
-        }
-
-        if (0 !== (self::MASK_OF_ACTION_LIST & $actionBit)
-            && $this->hasRoute('list')
-            && $this->hasAccess('list')
-        ) {
-            $buttonList['list'] = [
-                'template' => $this->getTemplateRegistry()->getTemplate('button_list'),
-            ];
-        }
-
+        $buttonList = $this->getInternalActionButtons($action, $object);
         $buttonList = $this->configureActionButtons($buttonList, $action, $object);
```

Comment:
We should avoid one-time variables, but I think it's better to use a more precise name instead of overriding the same variable here.
```suggestion
        $internalButtonList = $this->getInternalActionButtons($action, $object);
        $buttonList = $this->configureActionButtons($internalButtonList, $action, $object);
```

Reply:
This makes sense 👍